numpyro/infer/util.py

```diff
@@ -61,16 +61,27 @@ def log_density(model, model_args, model_kwargs, params, skip_dist_transforms=Fa
         if site['type'] == 'sample':
             value = site['value']
             intermediates = site['intermediates']
+            mask = site['mask']
+            scale = site['scale']
             if intermediates:
                 if skip_dist_transforms:
                     log_prob = site['fn'].base_dist.log_prob(intermediates[0][0])
                 else:
                     log_prob = site['fn'].log_prob(value, intermediates)
             else:
                 log_prob = site['fn'].log_prob(value)
+
+            # Minor optimizations
+            # XXX: note that this will not work for dynamic masks.
```

Comment:
I think this is fine. Users can provide ndarray for dynamic masks and this is JAX restriction, not NumPyro. I believe we can also have a check if mask is False to return early but it might not be important and can cause confusion (the benefit is to skip log prob computation for e.g. deterministic sites). WDYT?

Reply:
I think for that use case we can probably just use `handlers.block`? Otherwise, I'm not sure how that will work under JIT without resorting to `lax.cond`.

Reply:
Okay done.

Reply:
Oh, I didn't mean that but your solution works much better. Your solution is great!